source/sundials/copy.cc

```diff
@@ -56,6 +56,61 @@ namespace SUNDIALS
       return static_cast<std::size_t>(length);
     }
 
+
+    void
+    copy(LinearAlgebra::distributed::Vector<double> &dst, const N_Vector &src)
+    {
+      const IndexSet    is = dst.locally_owned_elements();
+      const std::size_t N  = is.n_elements();
+      AssertDimension(N, N_Vector_length(src));
+      dst.zero_out_ghost_values();
+      for (std::size_t i = 0; i < N; ++i)
+        {
+          dst[is.nth_index_in_set(i)] = NV_Ith_P(src, i);
```

Comment:
Why don't you use
```suggestion
          dst.local_element(i) = NV_Ith_P(src, i);
```
It seems unnatural to me to transform to global indices here and then back (on the exact same numbering).

Reply:
Changed!